packages/firestore/src/api/user_data_reader.ts

```diff
@@ -741,7 +741,11 @@ export function parseData(
   } else {
     // If context.path is null we are inside an array and we don't support
     // field mask paths more granular than the top-level array.
-    if (context.path) {
+    //
+    // If the input is undefined it can never participate in the fieldMask. With
+    // `ignoreUndefinedProperties` set to false, `parseScalarValue` will reject
+    // an undefined value.
+    if (context.path && input !== undefined) {
```

Comment:
It might be cleaner to pull the `undefined` check from `parseScalarValue` into this function. Something like:

```
if (lookslikeJson) { ... }
else if (input instanceof FieldValue) { ... }
else if (input == undefined && context.ignoreUndefinedProperties) {
    return null;
}  else { ... }
```

Optional.

Reply:
Good idea. Done.